### PR TITLE
refactor: Merge new upstream (go-git/go-git)

### PR DESCRIPTION
**Pull request by Nivl**
This PR pulls go-git 5.0 in. There are no API changes and it can be used as drop-in replacement: https://github.com/go-git/go-git/releases/tag/v5.0.0

The test failures are not on us and are an upstream issue. `version-matrix` is flaky, it sometimes passes, it sometimes fails.

Reviewers: We need to make extra sure we're not removing some of the features we added.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,11 +1,61 @@
+go-git Abstract's Fork
```
Comment (Nivl) [inline README.md:1]: I added some information about the fork

File: go.mod [hunk 2]
```diff
@@ -1,29 +1,26 @@
...
 	github.com/anmitsu/go-shlex v0.0.0-20161002113705-648efa622239 // indirect
 	github.com/armon/go-socks5 v0.0.0-20160902184237-e75332964ef5
-	github.com/davecgh/go-spew v1.1.1 // indirect
 	github.com/emirpasic/gods v1.12.0
 	github.com/flynn/go-shlex v0.0.0-20150515145356-3f9db97f8568 // indirect
 	github.com/gliderlabs/ssh v0.2.2
-	github.com/goabstract/go-git-fixtures v3.5.1+incompatible
+	github.com/go-git/gcfg v1.5.0
+	github.com/go-git/go-billy/v5 v5.0.0
+	github.com/go-git/go-git-fixtures/v4 v4.0.1
```
Comment (Nivl) [inline go.mod:11]: no need to have our own fork anymore \o/

**Review by davetakahashi-abstract — APPROVED** [review]
I don't see anything that looks like it's deleting our changes. 👍 

EDIT: Only thing I wonder is if we are going to fork off the new repo, is it easier to do that first, then apply our changes?

**Review by johncoder — APPROVED** [review]
Yeah, I don't see anything other than the few line items identified via GH Actions. Looks good.

**Comment by Nivl** [comment]
> EDIT: Only thing I wonder is if we are going to fork off the new repo, is it easier to do that first, then apply our changes?

We probably want to do a new fork at some point, but there's a few things to figure out first (like what will we name it since `goabstract/go-git` is already used and removing it will break go modules?). In the meantime, I'll merge this one so we have at least a base up-to-date here, then once we're ready to make a new fork it should be easy to do it (we should be able to just add this one as remote and pull the changes).

**Pull request merged**